runtime/src/iree-amd-aie/driver/hsa/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+iree_add_all_subdirs()
+
+iree_register_external_hal_driver(
+  NAME
+    amd-aie-hsa
+  DRIVER_TARGET
+    iree-amd-aie::driver::hsa::registration
+  REGISTER_FN
+    iree_hal_hsa_driver_module_register
+)
+
+include(iree_aie_utils)
+replace_string_in_file(
+  "${ROCR_RUNTIME_ROOT}/runtime/hsa-runtime/core/inc/amd_gpu_agent.h"
+  "static const" "const")
+
+if(NOT DEFINED HSA_API_HEADERS_ROOT)
+  set(HSA_API_HEADERS_ROOT "${ROCR_RUNTIME_ROOT}/runtime/hsa-runtime/inc")
+endif()
+
+iree_cc_library(
+  NAME
+    dynamic_symbols
+  HDRS
+    dynamic_symbols.h
+    status_util.h
+  TEXTUAL_HDRS
+    dynamic_symbol_tables.h
+  SRCS
+    dynamic_symbols.cc
+    hsa_headers.h
+    status_util.cc
+  INCLUDES
+    "${HSA_API_HEADERS_ROOT}"
+  DEPS
+    iree::base
+    iree::base::core_headers
+    iree::base::internal::dynamic_library
+)
+
+iree_cc_library(
+  NAME
+    hsa
+  HDRS
+    api.h
+  SRCS
+    api.h
```

Comment:
```suggestion
```

Reply:
you sure? all the other HALs have it like this https://github.com/iree-org/iree/blob/f51d4ea5c804b720836530a5bd71d7bffd5d924c/runtime/src/iree/hal/drivers/cuda/CMakeLists.txt#L17

Reply:
Here it's not needed: https://github.com/nod-ai/iree-amd-aie/blob/5559b049f81e844d02ab95f00fd17ae2211b686f/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/CMakeLists.txt#L38